File: app/src/main/java/com/winsun/fruitmix/file/view/viewmodel/FileItemViewModel.java
package com.winsun.fruitmix.file.view.viewmodel;

import android.databinding.ObservableBoolean;
import android.databinding.ObservableField;

import com.winsun.fruitmix.R;

/**
 * Created by Administrator on 2017/7/27.
 */

public class FileItemViewModel {

    public final ObservableBoolean selectMode = new ObservableBoolean(false);

    public final ObservableBoolean showFileIcon = new ObservableBoolean(true);

    public final ObservableField<Integer> fileIconBg = new ObservableField<>(R.drawable.round_circle);


}

File: app/src/main/java/com/winsun/fruitmix/newdesign201804/file/list/viewmodel/FileItemViewModel.kt
package com.winsun.fruitmix.newdesign201804.file.list.viewmodel

import android.databinding.ObservableBoolean
import android.databinding.ObservableField
import android.databinding.ObservableInt
import com.winsun.fruitmix.file.data.model.AbstractFile
import com.winsun.fruitmix.util.FileUtil

open class FileItemViewModel(val abstractFile: AbstractFile, var doHandleMoreBtnOnClick: () -> Unit = {}) {

    val isSelectMode = ObservableBoolean()

    val isSelected = ObservableBoolean()

    val folderName = ObservableField<String>()
    val fileTypeResID = ObservableInt()

    val fileFormatTime = ObservableField<String>()
    val fileFormatSize = ObservableField<String>()

    val showOfflineAvailableIv = ObservableBoolean(false)
    val showMoreBtn = ObservableBoolean(true)

    val isDisable = ObservableBoolean(false)

    fun init(){
        fileTypeResID.set(abstractFile.fileTypeResID)

        if (abstractFile.isFolder)
            fileFormatSize.set("")
        else
            fileFormatSize.set(FileUtil.formatFileSize(abstractFile.size))

        fileFormatTime.set(abstractFile.dateText)
        folderName.set(abstractFile.name)
    }


    fun handleMoreBtnOnClick() {
        doHandleMoreBtnOnClick()
    }

}